python/ir/bindings.cpp

```diff
@@ -0,0 +1,147 @@
+    REGISTER_CLASS(Const, ExprNode, m, "const", DataType, double)
+    REGISTER_CLASS(Cast, ExprNode, m, "cast", DataType, Expr)
+    REGISTER_CLASS(Get, ExprNode, m, "get", Expr, size_t)
+    REGISTER_CLASS(New, ExprNode, m, "new", vector<Expr>)
+
+    /* Math Operators for Nary Expressions */
+    py::enum_<MathOp>(m, "MathOp")
+        .value("_add", MathOp::ADD)
+        .value("_sub", MathOp::SUB)
+        .value("_mul", MathOp::MUL)
+        .value("_div", MathOp::DIV)
+        .value("_max", MathOp::MAX)
+        .value("_min", MathOp::MIN)
+        .value("_mod", MathOp::MOD)
+        .value("_sqrt", MathOp::SQRT)
+        .value("_pow", MathOp::POW)
+        .value("_abs", MathOp::ABS)
+        .value("_neg", MathOp::NEG)
+        .value("_ceil", MathOp::CEIL)
+        .value("_floor", MathOp::FLOOR)
+        .value("_lt", MathOp::LT)
+        .value("_lte", MathOp::LTE)
+        .value("_gt", MathOp::GT)
+        .value("_gte", MathOp::GTE)
+        .value("_eq", MathOp::EQ)
+        .value("_not", MathOp::NOT)
+        .value("_and", MathOp::AND)
+        .value("_or", MathOp::OR);
+
+    /* Nary Expressions */
+    REGISTER_CLASS(NaryExpr, ExprNode, m, "nary_expr", DataType, MathOp,
+                   vector<Expr>)
+    REGISTER_CLASS(UnaryExpr, NaryExpr, m, "unary_expr", DataType, MathOp, Expr)
+    REGISTER_CLASS(BinaryExpr, NaryExpr, m, "binary_expr", DataType, MathOp,
+                   Expr, Expr)
+
+    /* Math ops */
+    REGISTER_CLASS(Not, UnaryExpr, m, "_not", Expr)
+    REGISTER_CLASS(Abs, UnaryExpr, m, "_abs", Expr)
+    REGISTER_CLASS(Neg, UnaryExpr, m, "_neg", Expr)
+    REGISTER_CLASS(Sqrt, UnaryExpr, m, "_sqrt", Expr)
+    REGISTER_CLASS(Ceil, UnaryExpr, m, "_ceil", Expr)
+    REGISTER_CLASS(Floor, UnaryExpr, m, "_floor", Expr)
+    REGISTER_CLASS(Add, BinaryExpr, m, "_add", Expr, Expr)
+    REGISTER_CLASS(Sub, BinaryExpr, m, "_sub", Expr, Expr)
+    REGISTER_CLASS(Mul, BinaryExpr, m, "_mul", Expr, Expr)
+    REGISTER_CLASS(Div, BinaryExpr, m, "_div", Expr, Expr)
+    REGISTER_CLASS(Max, BinaryExpr, m, "_max", Expr, Expr)
+    REGISTER_CLASS(Min, BinaryExpr, m, "_min", Expr, Expr)
+    REGISTER_CLASS(Mod, BinaryExpr, m, "_mod", Expr, Expr)
+    REGISTER_CLASS(Pow, BinaryExpr, m, "_pow", Expr, Expr)
+    REGISTER_CLASS(Equals, BinaryExpr, m, "_eq", Expr, Expr)
+    REGISTER_CLASS(And, BinaryExpr, m, "_and", Expr, Expr)
+    REGISTER_CLASS(Or, BinaryExpr, m, "_or", Expr, Expr)
+
+    /* Logical Expressions */
+    REGISTER_CLASS(Implies, BinaryExpr, m, "implies", Expr, Expr)
+    REGISTER_CLASS(ForAll, NaryExpr, m, "for_all", Sym, Expr)
+    REGISTER_CLASS(Exists, NaryExpr, m, "exists", Sym, Expr)
+
+    /* Comparison Operators*/
+    REGISTER_CLASS(LessThan, BinaryExpr, m, "lt", Expr, Expr)
+    REGISTER_CLASS(GreaterThan, BinaryExpr, m, "gt", Expr, Expr)
+    REGISTER_CLASS(LessThanEqual, BinaryExpr, m, "lte", Expr, Expr)
+    REGISTER_CLASS(GreaterThanEqual, BinaryExpr, m, "gte", Expr, Expr)
```

Comment:
these ones should also start with underscore. to be consistent with all the other operations